ScoutSuite/providers/aws/resources/awslambda/functions.py

```diff
@@ -30,5 +29,30 @@ def _parse_function(self, raw_function):
         function_dict['version'] = raw_function.get('Version')
         function_dict['tracing_config'] = raw_function.get('TracingConfig')
         function_dict['revision_id'] = raw_function.get('RevisionId')
+
+        await self._add_role_information(function_dict, raw_function.get('Role'))
+        await self._add_access_policy_information(function_dict)
+
         return function_dict['name'], function_dict
 
+    async def _add_role_information(self, function_dict, role_id):
+        function_dict['role_arn'] = role_id
+        role_name = role_id.split("/")[-1]
+        function_dict['execution_role'] = await self.facade.iam.get_role_with_managed_policies(role_name)
+        # Make it easier to build rules based on policies attached to execution roles
+        statements = []
+        for policy in function_dict['execution_role']['policies']:
+            if 'Document' in policy and 'Statement' in policy['Document']:
+                statements += policy['Document']['Statement']
+        function_dict['execution_role']['policy_statements'] = statements
+
+    async def _add_access_policy_information(self, function_dict):
```

Comment:
I've simplified the implementation, so that we can display the whole policy in the report.